ios/LocationFullAccuracy/RNPermissionHandlerLocationFullAccuracy.m

```diff
@@ -0,0 +1,84 @@
+
+NSErrorDomain const RNPermissionHandlerLocationFullAccuracyDomain = @"RNPermissionHandlerLocationFullAccuracy";
+NS_ENUM(NSInteger) {
+  RNPermissionHandlerLocationFullAccuracyNoPurposeKey = 1,
+};
+
+@interface RNPermissionHandlerLocationFullAccuracy()
+
+@end
+
+@implementation RNPermissionHandlerLocationFullAccuracy
+
++ (NSArray<NSString *> * _Nonnull)usageDescriptionKeys {
+  return @[
+    @"NSLocationTemporaryUsageDescriptionDictionary"
+  ];
+}
+
++ (NSString * _Nonnull)handlerUniqueId {
+  return @"ios.permission.LOCATION_FULL_ACCURACY";
+}
+
++ (RNPermissionStatus)getAccuracyStatus:(CLLocationManager *)locationManager API_AVAILABLE(ios(14.0)) {
+  switch (locationManager.accuracyAuthorization) {
+    case CLAccuracyAuthorizationFullAccuracy:
+      return RNPermissionStatusAuthorized;
+    case CLAccuracyAuthorizationReducedAccuracy:
+    default:
+      return RNPermissionStatusNotDetermined;
+  }
+}
+
+- (void)checkWithResolver:(void (^ _Nonnull)(RNPermissionStatus))resolve
+                 rejecter:(void (__unused ^ _Nonnull)(NSError * _Nonnull))reject {
+  if (!CLLocationManager.locationServicesEnabled || CLLocationManager.authorizationStatus != RNPermissionStatusAuthorized) {
+    return resolve(RNPermissionStatusNotAvailable);
+  }
+
+  if (@available(iOS 14.0, *)) {
+    CLLocationManager *locationManager = [CLLocationManager new];
+    return resolve([RNPermissionHandlerLocationFullAccuracy getAccuracyStatus:locationManager]);
+  } else {
+    return resolve(RNPermissionStatusAuthorized);
+  }
+}
+
+- (void)requestWithResolver:(void (^ _Nonnull)(RNPermissionStatus))resolve
+                   rejecter:(void (^ _Nonnull)(NSError * _Nonnull))reject
+                  rationale:(NSDictionary *_Nullable)rationale {
+  if (!CLLocationManager.locationServicesEnabled || CLLocationManager.authorizationStatus != RNPermissionStatusAuthorized) {
+    return resolve(RNPermissionStatusNotAvailable);
+  }
+
+  if (@available(iOS 14.0, *)) {
+    CLLocationManager *locationManager = [CLLocationManager new];
+    NSString *purposeKey = [rationale objectForKey:@"temporaryPurposeKey"];
+
+    if (!purposeKey) {
+      return reject([NSError errorWithDomain:RNPermissionHandlerLocationFullAccuracyDomain
+                                        code:RNPermissionHandlerLocationFullAccuracyNoPurposeKey
+                                    userInfo:@{NSLocalizedDescriptionKey: @"temporaryPurposeKey is required to request LOCATION_FULL_ACCURACY"}]);
+    }
+
+    [locationManager requestTemporaryFullAccuracyAuthorizationWithPurposeKey:purposeKey
+                                                                  completion:^(NSError * _Nullable error) {
+      RNPermissionStatus status = [RNPermissionHandlerLocationFullAccuracy getAccuracyStatus:locationManager];
+
+      // Ignore errors due to full accuracy already being authorized
+      if (error && (error.code != kCLErrorPromptDeclined || status != RNPermissionStatusAuthorized)) {
```

Comment:
Isn't it `(error && error.code != kCLErrorPromptDeclined) || status != RNPermissionStatusAuthorized` ?

Reply:
No, a failure to get authorisation should still resolve the promise successfully.

What's happening here is that requestTemporaryFullAccuracyAuthorizationWithPurposeKey will error with kCLErrorPromptDeclined when permission is already granted, causing it not to show a prompt to the user.  When we see this error, and can confirm that authorisation is granted, the call is treated as a success instead.